specs/simple-serialize.md

```diff
@@ -54,43 +54,95 @@ For convenience we alias:
 
 We recursively define the `serialize` function which consumes an object `value` (of the type specified) and returns a bytestring of type `"bytes"`.
 
-*Note*: In the function definitions below (`serialize`, `hash_tree_root`, `signed_root`, etc.) objects implicitly carry their type.
+> *Note*: In the function definitions below (`serialize`, `hash_tree_root`, `signed_root`, etc.) objects implicitly carry their type.
 
-### `"uintN"`
+### Basic Types
+
+For basic types the `serialize` function is defined as follows.
+
+#### `"uintN"`
+
+A byte string of width  `N // 8` containing the little-endian encode integer.
 
 ```python
 assert N in [8, 16, 32, 64, 128, 256]
 return value.to_bytes(N // 8, "little")
 ```
 
-### `"bool"`
+#### `"bool"`
+
+* The byte `\x00` **if** the value is `False`
+* The byte `\x01` **if** the value is `True`
 
 ```python
 assert value in (True, False)
 return b"\x01" if value is True else b"\x00"
 ```
 
-### Vectors, containers, lists
+### Composite Types (Vectors, Containers and Lists)
 
-If `value` is fixed-size:
+The serialized representation of composite types is comprised of two binary segments.
 
-```python
-return "".join([serialize(element) for element in value])
-```
+* The first segment is *fixed size* for all types, containing the concatenation of *either* 
+    - The serialized representation of value for each of the *fixed size* types
+    - The `"uint32"` serialized offset where the serialized representation of the *variable sized* type is located in the second section.
```

Comment:
```suggestion
    - The `"uint32"` serialized offset where the serialized representation of the *variable sized* type is located in the second section relative to the beginning of the first section.
```

Long sentence, but without the addition it sounds like the offset is relative to the beginning of the second section.